app/views/questionnaire.py

```diff
@@ -200,15 +188,17 @@ def post_block(routing_path, schema, metadata, collection_metadata, answer_store
 @with_metadata
 @with_schema
 @full_routing_path_required
-def post_household_composition(routing_path, schema, metadata, answer_store, **kwargs):
+def post_household_composition(routing_path, schema, metadata, answer_store, **kwargs):  # pylint: disable=too-many-locals
     group_id = kwargs['group_id']
+    current_location = Location(group_id, 0, 'household-composition')
 
-    if _household_answers_changed(answer_store, schema):
-        _remove_repeating_on_household_answers(answer_store, schema)
+    answer_store_updater = AnswerStoreUpdater(current_location, schema, get_questionnaire_store(current_user.user_id, current_user.user_ik))
+    household_answer_changed = answer_store_updater.household_answers_changed(request.form.copy())
 
-    disable_mandatory = any(x in request.form for x in ['action[add_answer]', 'action[remove_answer]', 'action[save_sign_out]'])
+    if household_answer_changed:
+        answer_store_updater.remove_repeating_on_household_answers()
```

Comment:
Can the logic from these three lines all be dealt with with answer_store_updater?